File: flutter_ui_task/lib/models/doctor.dart
class Doctor {
  final int id;
  final String name;
  final String speciality;
  final String hospital;
  final double rating;
  final int reviewsCount;
  final String imageUrl;

  const Doctor({
    required this.id,
    required this.name,
    required this.speciality,
    required this.hospital,
    required this.rating,
    required this.reviewsCount,
    required this.imageUrl,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['id'] as int,
      name: json['name'] as String,
      speciality: json['speciality'] as String,
      hospital: json['hospital'] as String,
      rating: (json['rating'] as num).toDouble(),
      reviewsCount: json['reviewsCount'] as int,
      imageUrl: json['imageUrl'] as String? ?? '',
    );
  }
}

File: WebApplicationC#/WebApplicationC#/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;

namespace WebApplicationC.Models
{
    public class Doctor
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Speciality { get; set; } = string.Empty;

        [Required]
        public string Hospital { get; set; } = string.Empty;

        [Range(0, 5)]
        public double Rating { get; set; }

        public int ReviewsCount { get; set; }

        public string? ImageUrl { get; set; }

        // Lets us mark a subset of doctors as "recommended" on the
        // homepage without needing a separate table/endpoint later.
        public bool IsRecommended { get; set; } = true;
    }
}